packages/amplify-provider-awscloudformation/src/push-resources.ts

```diff
@@ -875,6 +839,92 @@ export const uploadTemplateToS3 = async (
   }
 };
 
+const createResourceObject = (resource: string, category: string) : {
+  category: string,
+  key: string
+} => ({
+  category: `${category}-${resource}`,
+  key: category + resource,
+});
+
+const getCategoryResources = (file : string, resourceDir : string) => {
+  const cloudFormationJsonPath = path.join(resourceDir, file);
+  const { cfnTemplate } = readCFNTemplate(cloudFormationJsonPath);
+  const categoryResources = Object.keys(cfnTemplate.Resources);
+  return categoryResources;
+};
+
+type EventMap = {
+  rootStackName: string,
+  envName: string,
+  projectName: string,
+  rootResources: {key: string, category: string}[],
+  eventToCategories: Map<string, string>,
+  categories: {name: string, size: number}[]
+}
+
+/**
+ Create an event map which will be used to map each incoming CloudFormation event into a category.
+ This is done so as to group events under progress bars for the root project as well as for each category.
+ */
+const createEventMap = (
+  context: $TSContext,
+  projectDetails: $TSObject,
+  resourcesToBeCreated: $TSAny,
+  resourcesToBeUpdated: $TSAny,
+): EventMap => {
+  let eventMap = {} as EventMap;
+
+  const { envName } = context.amplify.getEnvInfo();
+  const { projectName } = context.amplify.getProjectConfig();
+  const meta = stateManager.getMeta();
+  const rootStackName = meta.providers.awscloudformation.StackName;
+  // Setting up initial configurations.
+  eventMap.rootStackName = rootStackName;
+  eventMap.envName = envName;
+  eventMap.projectName = projectName;
+  eventMap.rootResources = [];
+  eventMap.eventToCategories = new Map();
+  eventMap.categories = [];
+
+  // Type script throws an error unless I explicitly convert to string
+  const resourcesUpdated = getAllUniqueCategories(resourcesToBeUpdated).map(item => `${item}`);
+  const resourcesCreated = getAllUniqueCategories(resourcesToBeCreated).map(item => `${item}`);
+
+  let categories = Object.keys(projectDetails.amplifyMeta);
+  categories = categories.filter(category => category !== 'providers');
+  categories.forEach(category => {
+    const resources = Object.keys(projectDetails.amplifyMeta[category]);
+    resources.forEach(resource => {
+      eventMap.rootResources.push(createResourceObject(resource, category));
+      handleCfnFiles(eventMap,
+        category, resource,
+        _.union(resourcesUpdated, resourcesCreated));
+    });
+  });
```

Comment:
This can be simplified since we don't need to store `categories` or `resources` in a variable.

```suggestion
  Object.keys(meta).forEach(category => {
    if (category === 'providers') {
      return;
    }
    Object.keys(meta[category]).forEach(resource => {
      eventMap.rootResources.push(createResourceObject(resource, category));
      handleCfnFiles(
        eventMap,
        category,
        resource,
        _.union(resourcesUpdated, resourcesCreated),
      );
    });
  });
```